### PR TITLE
Add GCC 12 & 14 to CI testing

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds GCC 12 & 14 to Linux_options.yml. It runs the main 5 jobs with GCC 12, and runs an additional, option-less job with GCC 14.

The `CFLAGS="-Wno-implicit-function-declaration -Wno-incompatible-pointer-types"` for jasper seems to be needed for GCC 14. I have not investigated it any further.

squash-n-merge

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
Once approved I'll update required tests

**Review by edwardhartnett — APPROVED** [review]

**Review by AlysonStahl-NOAA — APPROVED** [review]

**Comment by edwardhartnett** [comment]
@AlexanderRichert-NOAA do you want to do more work on this or is it ready to merge?

**Comment by AlexanderRichert-NOAA** [comment]
No this is good to go, I just removed the two required tests in the branch settings.

File: .github/workflows/Linux_options.yml [hunk 1]
```diff
@@ -18,17 +18,14 @@ concurrency:
...
-      CC: gcc
+    runs-on: ubuntu-24.04
 
     strategy:
 #      fail-fast: true
       matrix:
         config:
         - {
-            options: "-DUSE_IPOLATES=ON"
+            options: "-DUSE_IPOLATES=ON -DBLA_VENDOR=OpenBLAS"
```
Comment (edwardhartnett) [inline .github/workflows/Linux_options.yml:28]: Where did this new option come from?

**Pull request merged**